src/strategies/__tests__/spidStrategy.test.ts

```diff
@@ -0,0 +1,22 @@
+import { IDP_IDS, loadFromRemote } from "../spidStrategy";
+
+describe("loadFromRemote", () => {
+  it("should reject if the fetch of IdP metadata fails", () => {
+    const notExistingUrl = "http://0.1.2.3/index.html";
+    const result = loadFromRemote(notExistingUrl, IDP_IDS);
+    return expect(result).rejects.toEqual(expect.any(Error));
+  });
+
+  it("should reject an error if the fetch of IdP metadata returns no useful data", () => {
+    const wrongIdpMetadataUrl = "http://www.example.com";
+    const result = loadFromRemote(wrongIdpMetadataUrl, IDP_IDS);
+    return expect(result).rejects.toEqual(expect.any(Error));
+  });
+
+  it("should resolve with the fetched IdP options", () => {
+    const validIdpMetadataUrl =
+      "https://raw.githubusercontent.com/teamdigitale/io-backend/164984224-download-idp-metadata/test_idps/spid-entities-idps.xml";
```

Comment:
this must be a mocked http call, not a real one (as we're unit testing, not integration testing).
I suggest to use a fake URL and integrate https://github.com/nock/nock with a mocked response.